src/devices/machine/e05a30.cpp

```diff
@@ -197,6 +207,13 @@ uint8_t e05a30_device::read(offs_t offset)
 	LOG("%s: e05a30_r([0xC0%02x]): ", machine().describe_context(), offset);
 
 	switch (offset) {
+	case 0x00:
+		if (!machine().side_effects_disabled()) {
+			m_c000_read = BIT(m_c000_shift_register, 23) << 7;
```

Comment:
You don’t need this extra `m_c000_read` member.  Extracting a bit from the shift register doesn’t have side effects.  The extra member is redundant can just something that can go wrong.